### ISSUE TITLE
LoginForm_Edit shape unable to override the view

### ISSUE THREAD

**Issue by nbalu** [OrchardCMS/OrchardCore#15799]
It seems like that with the revamped login I'm unable to override the LoginForm.Edit.cshtml.

The LoginFormCredentials.cshtml works. Not sure what's wrong with the Edit.cshtml.

![image](https://github.com/OrchardCMS/OrchardCore/assets/7927881/058fcf68-d6b1-48ac-9d24-3e97b564e6f1)

using: 1.9.0-preview-18184

**Comment by Piedone** [comment]
There were breaking changes, see: https://docs.orchardcore.net/en/latest/docs/releases/1.9.0/#users-module Please follow that.

**Comment by nbalu** [comment]
Oh I've got it, just went through the pain of discovering hot the shape looks like and read through all sections.

So my key takeaways and fix me if I'm wrong here:
Login.cshtml -  still used key code snippet:
`@await DisplayAsync(Model) to render the corresponding shape`
![image](https://github.com/OrchardCMS/OrchardCore/assets/7927881/7af125d2-7463-4541-9656-9aad4f704801)

Content here:
https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Users/Views/LoginForm.Edit.cshtml
Where ContentZone get's rendered:
`@await DisplayAsync(Model.Content)`
![image](https://github.com/OrchardCMS/OrchardCore/assets/7927881/7647dade-61fb-4f69-b3c2-a0e2946f6f55)

Still I can override the LoginFormCredentials, but how should I name my LoginForm.Edit.cshtml to replace the original content?

Sorry for being a noobie here.

**Comment by Piedone** [comment]
You should be able to place a LoginForm.Edit.cshtml file into your module that depends on `OrchardCore.Users` to override it, or a theme but only if you configure this:

![image](https://github.com/OrchardCMS/OrchardCore/assets/1976647/a9bdceb6-7767-4517-9b50-4c232c7c0e64)

What exactly did you try?

Also see this that may help for future such cases: https://github.com/Lombiq/Helpful-Extensions?tab=readme-ov-file#shape-tracing-helpful-extensions

**Comment by nbalu2** [comment]
I've just created a webapp that depends on the CMS.Targets.

Enabled some features via AutoSetup + created a setup recipe.

But that's not a theme 😁 In my understanding. So everything got replaced except the "LoginForm.Edit.cshtml" thats the wierdness.

Thanks for the extension hint. Will take a look hopefully soon.

**Comment by Piedone** [comment]
Related: https://github.com/OrchardCMS/OrchardCore/pull/15701

**Comment by hishamco** [comment]
@MikeAlhayek could you please chime in

**Comment by MikeAlhayek** [comment]
Hisham, I think @Piedone handled it.

If someone is overriding the Login.chtml, they should adhere to these changes.  If you want to override Login.cshtml then you should render the LoginForm shape somewhere in your file. Then you can also override the sub views as needed.

@nbalu are you facing issues after reading @Piedone comments above?

**Comment by nbalu2** [comment]
Nope, so far I haven't checked the view resolutiin yet that Zoli suggested but my issue wasn't answered/solved.

What I did was to override:
Login.cshtml - whithin I called DisplayAsyc(Model)
LoginForm.Edit.cshtml - just removed the @T("Log in") to verify it works
LoginFormCredentials.cshtml

The first and last views I'm able to replace but LoginForm.Edit.cshtml is not overriden always the built in form gets generated.

I'm playing around with orchard and based ona design I'm creating a custom OIDC "server". Which is better then starting from scartch.

Though I need to touch every html that gets renderd to be able to style things.

**Comment by MikeAlhayek** [comment]
So when you call DisplayAsync(Model) in the login.cshtml you are rendering the LoginForm.Edit.cshtml.


The LoginForm.Edit.cshtml will render LoginFormCredentials.cshtml and other views.

If you are having issues, I suggest that you post the views you created to override the default templates so we can help you.

If you want to override any of the default views as @Piedone stated, you'll need to define the same views in a Theme project, or in a module that has physical reference to Orchard.User package.

**Comment by MikeAlhayek** [comment]
@nbalu did you figure it out or do you still think there is an issue?

**Comment by nbalu** [comment]
Hey Mike,

yes I've just solved it yet. But this was a wierd lesson for me:

On github the paths are:
src/OrchardCore.Modules/OrchardCore.Users/Views/Account/Login.cshtml - this is a view
src/OrchardCore.Modules/OrchardCore.Users/Views/LoginForm.Edit.cshtml - this is a shape

Within my repo:
src/OrchardCore.Webapp/Views/OrchardCore.Users/Account/Login.cshtml - this is working
src/OrchardCore.Webapp/Views/OrchardCore.Users/LoginForm.Edit.cshtml - this do not

Turns out that shape resolution works only on root level thus this worked:
src/OrchardCore.Webapp/Views/LoginForm.Edit.cshtml

--------------------------------

And logging at least shows that there's a shape, but still misleading as I've tried to map corresponding to it:
```
dbug: OrchardCore.DisplayManagement.Descriptors.ShapeTemplateStrategy.ShapeTemplateBindingStrategy[0]
      Binding 'Areas/OrchardCore.Users/Views/LoginForm.Edit.cshtml' as shape 'loginform_edit' for feature 'OrchardCore.Users'
```

It would be nice if the logger would show which path is treated as "Alternate" source and what's not.

--------------------------------

My issue with this way of coding is that it will be hard to distinct with what's overriden within orchard and what's my content definition. Plus I feel like "shape" should relate to content types not for "built-in" pages. 🤷 

Though I'm in the accomodation phase where I try to oppress my coding instincts before judge any framework or authoring logic of yours before being experienced enough to argue with you 😆 

This was a true RFTM issue of mine. And I really appriciate the responsiveness of the community of yours. It's great to see how helpful You are and respect what you are doing here. Kudos
